js/block.js

```diff
@@ -3656,6 +3656,17 @@ class Block {
                     '" />';
                 labelElem.classList.add("hasKeyboard");
                 this.label = docById("numberLabel");
+
+                    // Add an input event listener to enforce the 4-digit limit
+        this.label.addEventListener("input", function () {
+        const inputValue = this.value;
+        if (inputValue.length > 6) {
+            this.errorText = docById("errorText");
+            this.errorText.classList.add("show");
+            this.errorText.innerHTML = "Do not put input greater than 6 digits";
```

Comment:
The error message will need to be translated, so use _().
e.g., _("Do not use a number of more than 6 digits.")